addons/bitbucket/apps.py

```diff
@@ -21,12 +21,13 @@ def bitbucket_hgrid_data(node_settings, auth, **kwargs):
     node = node_settings.owner
     if node.is_public and not node.is_contributor(auth.user):
         try:
-            connection.repo(node_settings.user, node_settings.repo)
+            repo = connection.repo(node_settings.user, node_settings.repo)
         except NotFoundError:
             # TODO: Add warning message
             logger.error('Could not access Bitbucket repo')
             return None
-
+        if not repo:
+            return None
```

Comment:
Can this be moved into the "try" branch - just my personal preference.

Reply:
Nice catch. I will assign a repo to avoid the potential problem before the "try". But I always think we should put as less logic inside a "try" branch as possible. Correct me if I am wrong.

Reply:
Actually no need to assign it as the logic will either get a value for repo or return None.

Reply:
if you don't want that in the "try" branch that's fine, as I said, it's my personal preference.